modules/andes-core/client/src/main/java/org/wso2/andes/jndi/PropertiesFileInitialContextFactory.java

```diff
@@ -180,6 +184,41 @@ protected void createConnectionFactories(Map data, Hashtable environment) throws
         }
     }
 
+    /**
+     * Resolve carbon secure vault encrypted properties.
+     */
+    private static void resolveEncryptedProperties(Hashtable environment) {
+
+        if (environment != null) {
+
+            Map<String, String> envVariables = (Hashtable<String, String>) environment;
+            Properties properties = convertToProperties(envVariables);
+            SecretResolver secretResolver = SecretResolverFactory.create(properties);
+
+            for (Object key : envVariables.keySet()) {
+
+                if (secretResolver != null && secretResolver.isInitialized()) {
+                    if (secretResolver.isTokenProtected(key.toString())) {
+                        environment.put(key.toString(), secretResolver.resolve(key.toString()));
+                    }
+                }
+
+            }
+
+        }
+    }
+
+    private static Properties convertToProperties(Map map) {
```

Comment:
In practice we document all the methods ( in including parameters).
Shall we do that here too?